java/integTest/src/test/java/glide/TestUtilities.java

```diff
@@ -76,4 +80,53 @@ public static Map<String, String> parseInfoResponseToMap(String serverInfo) {
         return RedisClusterClientConfiguration.builder()
                 .address(NodeAddress.builder().port(CLUSTER_PORTS[0]).build());
     }
+
+    /**
+     * Deep traverse and compare two objects, including comparing content of all nested collections
+     * recursively.
+     *
+     * @apiNote Maps and Set comparison ignores their order, regardless of `orderMatters` argument.
+     *     Their entries could be reordered, but values stored in them compared according to this
+     *     parameter.
+     */
+    public static void assertDeepEquals(Object expected, Object actual, boolean orderMatters) {
```

Comment:
when would we want to use orderMatters == false?

Reply:
No ideas actually

Reply:
I see that it's really only for the key comparison.  

Let's remove it if we remove the key checking below - and we can add it in if it's needed later.